providers/harness/pkg/provider.go

```diff
@@ -0,0 +1,310 @@
+package harness
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/harness/ff-golang-server-sdk/client"
+	harness "github.com/harness/ff-golang-server-sdk/client"
+	"github.com/harness/ff-golang-server-sdk/evaluation"
+	"github.com/harness/ff-golang-server-sdk/types"
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+)
+
+const providerNotReady = "Provider not ready"
+const generalError = "general error"
+
+type Provider struct {
+	providerConfig ProviderConfig
+	harnessClient  *client.CfClient
+	status         of.State
+}
+
+func NewProvider(providerConfig ProviderConfig) (*Provider, error) {
+	provider := &Provider{
+		providerConfig: providerConfig,
+		status:         of.NotReadyState,
+	}
+	return provider, nil
+}
+
+func (p *Provider) Init(evaluationContext of.EvaluationContext) error {
+	harnessClient, err := harness.NewCfClient(p.providerConfig.SdkKey, p.providerConfig.Options...)
+	if err != nil {
+		p.status = of.ErrorState
+	} else {
+		p.status = of.ReadyState
+		p.harnessClient = harnessClient
+	}
+	return err
```

Comment:
Does harness include any sort of change events? We don't need it for first release, but we could maybe make an issue for it.

Reply:
I didn't see events in Harness GO SDK which can be mapped to provider events. If so then it can be address as a follow-up issue yes.

Reply:
@liran2000 I think this is something we could add into the golang SDK.  Currently we expose a func to tell you if it is initialised.  We debated about exposing a channel to send the event on, but opted not to.  I think what we can do is make it a option to pass into the constructor, where you can pass you're own channel in and get a single when we successfully init

Reply:
I think that would be a cool feature. Personally I've always found change events in SDKs useful.